addons/phone_validation/models/mail_thread_phone.py

```diff
@@ -47,14 +47,20 @@ class PhoneMixin(models.AbstractModel):
     phone_mobile_search = fields.Char("Phone/Mobile", store=False, search='_search_phone_mobile_search')
 
     def _search_phone_mobile_search(self, operator, value):
+        value = value.strip() if isinstance(value, str) else value
         phone_fields = [
             fname for fname in self._phone_get_number_fields()
             if fname in self._fields and self._fields[fname].store
         ]
         if not phone_fields:
             raise UserError(_('Missing definition of phone fields.'))
 
-        value = value.strip()
+        # search if phone/mobile is set or not
+        if (value is True or not value) and operator in ['=', '!=']:
+            final_operator = '=' if (operator == '=' or value) else '!='
+            op = '&' if final_operator == '=' else '|'
+            return [op] * (len(phone_fields) - 1) + [(phone_field, final_operator, False) for phone_field in phone_fields]
```

Comment:
a bit more readable with expression.AND / expression.OR imo
```suggestion
        # search if phone/mobile is set or not
        if (value is True or not value) and operator in ('=', '!='):
            if value:
                # inverse the operator
                operator = '=' if operator == '!=' else '!='
            op = expression.AND if operator == '=' else expression.OR
            return op([[(phone_field, operator, False)] for phone_field in phone_fields])
```

(and annoying std, tuple better than list for this case)